cli_tools/gce_ovf_import/ovf_importer/ovf_importer.go

```diff
@@ -313,14 +321,129 @@ func (oi *OVFImporter) importDiskFilesToImages() error {
 		return err
 	}
 
+	oi.params.OsID = osIDValue
+
 	oi.machineTypeString, err = oi.getMachineType(ovfDescriptor, *oi.params.Project, oi.params.Zone)
 	if err != nil {
 		return err
 	}
 
 	oi.Logger.User(fmt.Sprintf("Will create instance of `%v` machine type.", oi.machineTypeString))
 
-	return oi.importDisks(osIDValue, &diskInfos)
```

Comment:
The part of code i added here is main consists of 2 main functionalities:
1) create data disks for all VMDK data files.
2) import the boot disk VMDK file into image, it is be called after the data disks are created.
I did't use multiImage importer as I need to create the data disks first then import boot image that will use these disks, what do u think?

Reply:
I don't think `multiImageImporter` is used anymore after your change, and it looks like the code you've added to this file is conceptually similar to the goal of `multiImageImporter`: Given multiple VMDK files, import them to GCP in parallel.

You'll want to put your new code behind an interface, and call that interface from `ovf_importer.go`. If you want to use `multiImageImporter` as that interface, I'm happy with that. If you want to delete it and create a different interface, that's great too.

The reason for this refactoring is to avoid adding detailed logic to `ovf_importer.go`. This file is really large, so testing detailed logic is difficult given the number of dependencies that need to be constructed. By putting your new code behind a dedicated interface, you have a minimal number of dependencies, and testing becomes easier. Read through the code in `multi_image_importer`; it splits functionality into files such that it's easy to test corner cases using unit tests.